mwccgap/mwccgap.py

```diff
@@ -258,9 +245,19 @@ def compile_file(
     mwcc_path: Path,
     use_wibo: bool,
     wibo_path: Path,
+    source_encoding: Optional[str],
 ):
     with tempfile.TemporaryDirectory() as temp_dir:
         o_file = Path(temp_dir) / "result.o"
+
+        if source_encoding != None:
+            encoded_c_file = tempfile.NamedTemporaryFile(suffix=".c", dir=c_file.parent)
```

Comment:
I opted not to use the `with` syntax here as it makes the function kinda clunky. IIRC Python does not garbage collect variables before the function returns, so the file-like object returned by `NamedTemporaryFile` is guaranteed to be alive for the duration of the function